react-native/ios/RealmReact/RealmReact.mm

```diff
@@ -55,20 +56,21 @@ - (JSGlobalContextRef)ctx;
     id rctJSContext = object_getIvar(executor, contextIvar);
     if (!rctJSContext && create) {
         Class RCTJavaScriptContext = NSClassFromString(@"RCTJavaScriptContext");
-        NSMethodSignature *signature = [RCTJavaScriptContext instanceMethodSignatureForSelector:@selector(initWithJSContext:)];
-        assert(signature);
 
-        // React Native 0.14+ expects a JSContext here, but we also support 0.13.x, which takes a JSGlobalContextRef.
-        if (!strcmp([signature getArgumentTypeAtIndex:2], "@")) {
-            JSContext *context = [[JSContext alloc] init];
-            rctJSContext = [[RCTJavaScriptContext alloc] initWithJSContext:(void *)context];
+        NSMethodSignature *signature = [RCTJavaScriptContext instanceMethodSignatureForSelector:@selector(initWithJSContext:onThread:)];
+        if (signature) {
+            // for RN 0.28.0+
+            rctJSContext = [executor context];
```

Comment:
In my opinion this is too hackish. You are deducing that when there is support for `initWithJSContext:onThread:` there is also `[executor context]`. This can break easily in future RN release.

I'd suggest to move `[executor context]` to the top of the function, because it should be **the** way to ask for context. See my [commit](https://github.com/heracek/realm-js/commit/fec5ab408f0b6d4ca18306bf105af96539d9e72c).

`[executor context]` was introduces in [this commit](https://github.com/facebook/react-native/commit/a5d82c282336c80addf1225db1346c764b6794f7#diff-b2c406c59f5d64abb6f8eca647b69e8aR218) and it seems that it [remained the same](https://github.com/facebook/react-native/blame/master/React/Executors/RCTJSCExecutor.mm#L376) until [this commit](https://github.com/facebook/react-native/commit/1ebd9c5deaa453a56e73f4fc34165f08de18bd79#diff-24171be54ca5e6987e6e9b2f870fe8a3L307) removed lazy creation of context (`if (!_context) {...}`).

Commit removing lazy creation of context is only on master for now. And maybe it would mitigate any problems with context being `nil`, because context is created as part of `setUp` method.

There is also other hack to ivar in my commit, but I don't know how now to get `javaScriptThread` other way.

Whet do you think?